frontend/src/App.tsx

```diff
@@ -0,0 +1,144 @@
+import ConnectionString, {ValidationMessage} from "@components/ConnectionString";
+import Editor from "@components/Editor";
+import Limit from "@components/Limit";
+import Profile from "@components/Profile";
+import StatusBar from "@components/StatusBar";
+import {LimitModel} from "@models/LimitModel";
+import {ProfileModel} from "@models/ProfileModel";
+import {ConnectionService} from "@services/ConnectionService";
+import {ProviderService} from "@services/ProviderService";
+import * as monaco from "monaco-editor";
+import React, {BaseSyntheticEvent, useEffect, useState} from "react";
+import {ProfilesApi} from "@api/ProfilesApi";
+import {LimitsApi} from "@api/LimitsApi";
+
+
+function App() {
+  const [connectionString, setConnectionString] = useState<string | null>(null);
+  const [connectionStringValidationMessage, setConnectionStringValidationMessage] = useState<ValidationMessage | null>(null);
+
+  const [selectedProvider, setSelectedProvider] = useState<string | null>(null);
+  const [selectedProfile, setSelectedProfile] = useState<ProfileModel | null>(null);
+  const [selectedLimit, setSelectedLimit] = useState<LimitModel | null>(null);
+
+  const [providers, setProviders] = useState<string[]>([]);
+  const [profiles, setProfiles] = useState<ProfileModel[]>([]);
+  const [limits, setLimits] = useState<LimitModel[]>([]);
+
+  function selectProfile(profileType: string) {
+    const profile: ProfileModel = profiles.find(p => p.type === profileType)!;
+    setSelectedProfile(profile);
+  }
+
+  function selectLimit(limitType: string) {
+    const limit: LimitModel = limits.find(p => p.type === limitType)!;
+    setSelectedLimit(limit);
+  }
+
+  function selectProvider(provider: string) {
+    setSelectedProvider(provider);
+    ProviderService.saveCurrent(provider);
+  }
+
+  function execute(event: BaseSyntheticEvent) {
+    event.preventDefault();
+
+    ExecutionApi.run({
+      provider: selectedProvider,
+      connectionString: connectionString,
+      script: monaco.editor.getEditors()[0].getValue(),
+      profile: selectedProfile,
+      limit: selectedLimit
+    }).then(() => {
+      /* TODO: processing result */
+    });
+  }
+
+  function testConnectionString() {
+    ConnectionService.test(selectedProvider, connectionString)
+      .then(message => setConnectionStringValidationMessage(message));
+  }
+
+  function loadProviders(): void {
+    ProvidersApi
+      .getAll()
+      .then(providers => setProviders(providers));
+
+    const currentProvider = ProviderService.getCurrent();
+    setSelectedProvider(currentProvider);
+  }
+
+  function loadProfiles(): void {
+    ProfilesApi
+      .getAll()
+      .then(profiles => setProfiles(profiles));
+  }
+
+  function loadLimits(): void {
+    LimitsApi
+      .getAll()
+      .then(limits => setLimits(limits));
+  }
+
+  useEffect(() => {
+    loadProviders();
+    loadProfiles();
+    loadLimits();
+  }, []);
+
+  return (
+    <div className="container-fluid px-0 px-xl-5">
```

Comment:
I would advise using this tool https://reactjs.org/docs/strict-mode.html to detect side effects, etc.

Reply:
React.StrictMode is one level higher in main.tsx, is that enough?